src/solver/expressions/include/antares/solver/expressions/visitors/EvaluationContext.h

```diff
@@ -1,23 +1,35 @@
 #pragma once
 
 #include <map>
-#include <string>
 
 namespace Antares::Solver::Visitors
 {
+template<class KeyType, class ValueType>
 class EvaluationContext
 {
 public:
     EvaluationContext() = default;
-    explicit EvaluationContext(std::map<std::string, double> parameters,
-                               std::map<std::string, double> variables);
 
-    double getVariableValue(const std::string&) const;
-    double getParameterValue(const std::string&) const;
+    explicit EvaluationContext(std::map<KeyType, ValueType> parameters,
+                               std::map<KeyType, ValueType> variables):
+        parameters_(std::move(parameters)),
+        variables_(std::move(variables))
+    {
+    }
+
+    ValueType getVariableValue(const KeyType& key) const
+    {
+        return variables_.at(key);
+    }
+
+    ValueType getParameterValue(const KeyType& key) const
+    {
+        return parameters_.at(key);
+    }
 
 private:
-    std::map<std::string, double> parameters_;
-    std::map<std::string, double> variables_;
+    std::map<KeyType, ValueType> parameters_;
+    std::map<KeyType, ValueType> variables_;
```

Comment:
In my opinion you don't need a template here. You can just declare a single
`std::map<Node*, TimeIndex>`.

For the use cases, you just need to populate this map for all leaves (parameters, literal, etc.)